xonsh/history.py

```diff
@@ -290,6 +290,15 @@ def append(self, cmd):
         hf : HistoryFlusher or None
             The thread that was spawned to flush history
         """
+        opts = builtins.__xonsh_env__.get('HISTCONTROL', set())
+        if ('ignoredups' in opts and len(self.buffer)
```

Comment:
Can you make the comparison explicit here, ie `len(self.buffer) > 0`, please?  Also, why are you testing the length of the buffer rather than the length of history?  So shouldn't this be `len(self) > 0`?